tools/codemod/src/utils/TransformWidget.ts

```diff
@@ -0,0 +1,135 @@
+import { JSCodeshift, ASTPath, JSXElement, MemberExpression, ObjectExpression, ObjectProperty, Collection } from "jscodeshift";
+import { ObjectExpressionCollection, useObjectExpression } from "./ObjectExpression";
+import { ConfigExpression, getObjectProperty, jsxElementToUniFormat, uniFormatToObjectExpression } from "./TransformUtils";
+
+export function transformWidget(j: JSCodeshift, widget: ASTPath<JSXElement>): ConfigExpression { // TODO: registerMethod
+  const uni = jsxElementToUniFormat(j, widget);
+  const widgetObj = uniFormatToObjectExpression(j, uni);
+
+  // Add ---------------------------
```

Comment:
Remove these comments, most of the code should be self describing, w/ some inline comments for complicated cases.